aioamqp/__init__.py

```diff
@@ -58,6 +59,14 @@ def connect(host='localhost', port=None, login='guest', password='guest',
         factory, host, port, **create_connection_kwargs
     )
 
+    # these 2 flags *may* show up in sock.type. They are only available on linux
+    # see https://bugs.python.org/issue21327
+    nonblock = getattr(socket, 'SOCK_NONBLOCK', 0)
+    cloexec = getattr(socket, 'SOCK_CLOEXEC', 0)
+    sock = transport.get_extra_info('socket')
+    if sock is not None and (sock.type & ~nonblock & ~cloexec) == socket.SOCK_STREAM:
```

Comment:
Isn't AMQP always over TCP? I don't think I understand what you are checking here?